ansible_rulebook/action/run_job_template.py

```diff
@@ -0,0 +1,156 @@
+                    logger.info(
+                        "Previous run_job_template failed. Retry %d of %d",
+                        i,
+                        retries,
+                    )
+                controller_job = await job_template_runner.run_job_template(
+                    self.name,
+                    self.organization,
+                    self.job_args,
+                )
+                if controller_job["status"] != "failed":
+                    break
+        except (ControllerApiException, JobTemplateNotFoundException) as ex:
+            logger.error(ex)
+            controller_job = {}
+            controller_job["status"] = "failed"
+            controller_job["created"] = run_at()
+            controller_job["error"] = str(ex)
+
+        self.controller_job = controller_job
+        await self._post_process()
+
+    async def _post_process(self) -> None:
+        a_log = {
+            "job_template_name": self.name,
+            "organization": self.organization,
+            "job_id": self.job_id,
+            "status": self.controller_job["status"],
+            "run_at": self.controller_job["created"],
+            "url": self._controller_job_url(),
+            "matching_events": self.helper.get_events(),
+        }
+        if "error" in self.controller_job:
+            a_log["message"] = self.controller_job["error"]
+            a_log["reason"] = {"error": self.controller_job["error"]}
+
+        await self.helper.send_status(a_log)
+        set_facts = self.action_args.get("set_facts", False)
+        post_events = self.action_args.get("post_events", False)
+
+        if set_facts or post_events:
+            ruleset = self.action_args.get(
+                "ruleset", self.helper.metadata.rule_set
+            )
+            logger.debug("set_facts")
+            facts = self.controller_job.get("artifacts", {})
+            if facts:
+                facts = self.helper.embellish_internal_event(facts)
+                logger.debug("facts %s", facts)
+                if set_facts:
+                    lang.assert_fact(ruleset, facts)
+                if post_events:
+                    lang.post(ruleset, facts)
+            else:
+                logger.debug("Empty facts are not set")
+
+    async def _job_start_event(self):
+        await self.helper.send_status(
+            {
+                "run_at": run_at(),
+                "matching_events": self.helper.get_events(),
+                "action": self.helper.action,
+                "hosts": ",".join(self.helper.control.hosts),
+                "name": self.name,
+                "job_id": self.job_id,
+                "ansible_rulebook_id": settings.identifier,
+            },
+            obj_type="Job",
+        )
+
+    def _controller_job_url(self) -> str:
+        if "id" in self.controller_job:
+            return (
+                f"{job_template_runner.host}/#/jobs/"
+                f"{self.controller_job['id']}/details"
```

Comment:
This should have trailing slash, don't?

Reply:
@Alex-Izquierdo This is a url value that we store in our DB allowing users to access the related job which happens from the browser.